django/db/backends/mysql/base.py

```diff
@@ -263,13 +288,25 @@ def get_new_connection(self, conn_params):
         return conn
 
     def init_connection_state(self):
+        init_statements = []
+
         if self.features.is_sql_auto_is_null_enabled:
+            # SQL_AUTO_IS_NULL controls whether an AUTO_INCREMENT column on
+            # a recently inserted row will return when the field is tested
+            # for NULL. Disabling this brings this aspect of MySQL in line
+            # with SQL standards.
+            init_statements.append('SET SQL_AUTO_IS_NULL = 0')
+
+        if self.tx_isolation_level != 'repeatable read':
```

Comment:
Oh, it might already be set on server side in config (`tx_isolation` setting) to something other than `repeatable read`. We should send it every time from the Django setting in that case.

We might also want an option that avoids the overhead of the extra `SET` command every time such as `None` = use server value ?